Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -2889,9 +2889,17 @@ void HTMLMediaElement::changeNetworkStateFromLoadingToIdle()
 void HTMLMediaElement::mediaPlayerReadyStateChanged()
 {
     if (isSuspended()) {
-        queueTaskKeepingObjectAlive(*this, TaskSource::MediaElement, [this] {
-            mediaPlayerReadyStateChanged();
-        });
+        if (!m_isChangingReadyStateWhileSuspended) {
+            m_isChangingReadyStateWhileSuspended = true;
+            m_remainingReadyStateChangedAttempts.store(128);
+        }
+
+        if (m_remainingReadyStateChangedAttempts.exchangeSub(1)) {
+            queueTaskKeepingObjectAlive(*this, TaskSource::MediaElement, [this] {
+                mediaPlayerReadyStateChanged();
+            });
+        } else
+            ALWAYS_LOG(LOGIDENTIFIER, "Possible infinite recursion detected, bailing out");
```

Comment:
I would make this a warning and add a comment at the beginning of this block of code.

Reply:
If you add a warning, we'll surely get bug reports complaining about the warning. We should only print warnings when something unexpected happens. We know this condition is expected and certain to occur regularly.

(I have a feeling there must be a better solution for this bug, but I know very little about multimedia and don't have any suggestions.)

Reply:
So you're fine with `ALWAYS_LOG`?

And yes, this is a workaround. I'm not entirely satisfied about this either.
I suspect this isn't an issue on mac ports because they run media stuff in GPUP, so with different runtime behaviour, and with working process suspension support.

Reply:
Personally, I would add a FIXME and not print anything (unless you're OK with receiving bug reports when it prints).